src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs

```diff
@@ -373,7 +373,7 @@ private static LibraryDependency ToPackageLibraryDependency(LegacyCSProjPackageR
             {
                 LibraryRange = new LibraryRange(
                     name: item.Name,
-                    versionRange: new VersionRange(new NuGetVersion(item.Version), originalString: item.Version),
+                    versionRange: VersionRange.Parse(item.Version),
```

Comment:
It might be helpful to wrap it and include the package id with it, up to you on that.